Step Aliasing: allow steps to be invoked by multiple names

steven-terrana opened the pull request:
# PR Details

The [new docs for this feature](https://github.com/steven-terrana/templating-engine-plugin/blob/step-alias/docs/modules/library-development/pages/step_aliasing.adoc) are probably the fastest way to get context

## Description

Adds a `@StepAlias` annotation that allow steps to be invoked by multiple names. 

## How Has This Been Tested

Unit Tests

## Types of Changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Added Unit Testing
- [x] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [x] I have added the appropriate label for this PR
- [x] If necessary, I have updated the documentation accordingly.
- [x] All new and existing tests passed.

cokieffebah commented on src/main/groovy/org/boozallen/plugins/jte/init/primitives/injectors/StepAlias.groovy:
```diff
@@ -0,0 +1,48 @@
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+package org.boozallen.plugins.jte.init.primitives.hooks
```
should this file be in the equivalent package/directory, src/main/groovy/org/boozallen/plugins/jte/init/primitives/hooks/StepAlias.groovy, instead of src/main/groovy/org/boozallen/plugins/jte/init/primitives/injectors/StepAlias.groovy

cokieffebah commented on docs/modules/library-development/pages/step_aliasing.adoc:
```diff
@@ -0,0 +1,163 @@
+Previously, when writing libraries such as this, common logic around tool versioning, error checking, etc would have to be either duplicated across multiple libraries.
+Sometimes, a generic invoking step would be created and accept the "phase" as a method argument from other library steps.
+
+Step Aliasing simplifies these types of setups.
```
what is the behavior for StepAlias on other methods in the file? 
```
@StepAlias(value = "build", keepOriginal = true) <1>
void call(){
    println "running as either build() or generic()"
}

@StepAlias(value = "build2") 
void otherBuild(){
    println "running as either build2() or otherBuild()"
}
```

steven-terrana commented:
PR updated based on feedback

1. cleanup from #210 integrated
2. added `stepContext` variable to steps
3. fail if more than 1 `@StepAlias` found in a step
4. added test for the intended use case
5. added the flag conditional in `customizeImports`
